File: app/src/main/java/com/example/todonotesapp/adapter/NotesAdapter.java
package com.example.todonotesapp.adapter;


import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.todonotesapp.ClickListeners.ItemClickListener;
import com.example.todonotesapp.R;
import com.example.todonotesapp.model.Notes;

import java.util.List;

public class NotesAdapter extends RecyclerView.Adapter<NotesAdapter.ViewHolder> {

    private List<Notes> listNotes;

    private ItemClickListener itemClickListener;


    public NotesAdapter(List<Notes> list, ItemClickListener itemClickListener) {
        this.listNotes = list;
        this.itemClickListener = itemClickListener;
    }

    @NonNull
    @Override
    public NotesAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.notes_adapter_layout, parent,false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull NotesAdapter.ViewHolder holder, int position) {
        // bind the data from list to the viewholder

        final Notes notes = listNotes.get(position);

        String title = notes.getTitle();
        String description = notes.getDescription();
        holder.textViewTitle.setText(title);
        holder.textViewDescription.setText(description);

        // setting onclick listener on ITEMs
        holder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                itemClickListener.onClick(notes);
            }
        });
    }

    @Override
    public int getItemCount() {

        return listNotes.size();
    }
    static class ViewHolder extends RecyclerView.ViewHolder {
    TextView textViewTitle, textViewDescription;
        public ViewHolder(@NonNull View itemView) {

            super(itemView);
            textViewTitle = itemView.findViewById(R.id.textViewTitle);
            textViewDescription = itemView.findViewById(R.id.textViewDescription);
        }
    }
}

File: app/src/main/java/com/example/todonotesapp/adapter/NotesAdapter.kt
package com.example.todonotesapp.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.CheckBox
import android.widget.CompoundButton
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.todonotesapp.ClickListeners.ItemClickListener
import com.example.todonotesapp.R
import com.example.todonotesapp.db.Notes

class NotesAdapter(private val list : List<Notes>, private val itemClickListener: ItemClickListener) :RecyclerView.Adapter<NotesAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.notes_adapter_layout, parent, false)
        return ViewHolder(view)
    }

    override fun getItemCount(): Int {
        return list.size
    }

//    override fun onBindViewHolder(holder: NotesAdapter.ViewHolder, position: Int) {
        override fun onBindViewHolder(holder: NotesAdapter.ViewHolder, position: Int) {

        val notes = list[position] // int java -> list.get(position)
        val title = notes.title
        val description = notes.description

        holder.textViewTitle.text = title
        holder.textViewDescription.text = description

        holder.checkBoxMarkStatus.isChecked = notes.isTaskCompleted

    holder.itemView.setOnClickListener (object : View.OnClickListener {
            override fun onClick(v: View?) {
            itemClickListener.onUpdate(notes)
        }

    })

    holder.checkBoxMarkStatus.setOnCheckedChangeListener(object : CompoundButton.OnCheckedChangeListener{
        override fun onCheckedChanged(buttonView: CompoundButton?, isChecked: Boolean) {
            notes.isTaskCompleted = isChecked
            itemClickListener.onUpdate(notes)
        }

    })
}

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val textViewTitle : TextView = itemView.findViewById(R.id.textViewTitle)
        val textViewDescription : TextView = itemView.findViewById(R.id.textViewDescription)
        val checkBoxMarkStatus : CheckBox = itemView.findViewById(R.id.checkboxMarkStatus)
    }

}